lib/gui.py

```diff
@@ -413,7 +415,33 @@ def forward_and_insert_hash_result(input, buf_size, algorithm, output):
 
 
 def make_hash_cracker(main_window):
-    pass
+    hash_cracker_window, hash_cracker_frame = make_top_level(main_window, "Hash Cracker")
+    hash_to_crack = tk.StringVar(value=None)
+    wordlist_path = tk.StringVar(value="Default")
+    hash_algorithm = tk.StringVar(value="SHA256")
+
+    def forward_and_crack_hash(hash, wordlist, hash_algorithm):
+        result = crack(hash, wordlist, hash_algorithm)
+        result_hash_crack.delete("1.0", "end-1c")
+        result_hash_crack.insert(tk.END, str(result))
+
+    widgets = [
+        tk.Label(hash_cracker_frame, text="Hash to crack"),
+        EntryWithPlaceholder(hash_cracker_frame, "", textvariable=hash_to_crack),
+        tk.Label(hash_cracker_frame, text="Wordlist Path"),
+        EntryWithPlaceholder(hash_cracker_frame, "", textvariable=wordlist_path),
+        tk.Label(hash_cracker_frame, text="Hash Type"),
+        EntryWithPlaceholder(hash_cracker_frame, "", textvariable=hash_algorithm),
+        tk.Button(
+            hash_cracker_window,
+            text="Crack the hash",
+            command=lambda: forward_and_crack_hash(hash_to_crack.get(), wordlist_path.get(), hash_algorithm.get()),
+        ),
+    ]
+    pack_widgets(widgets)
+    result_hash_crack = tk.Text(hash_cracker_frame)
+    result_hash_crack.pack()
+    copy_button = tk.Button(hash_cracker_frame, text="Copy cracked hash")
```

Comment:
When a hash is cracked it is then represented as a plain text string not a hash. `Copy cracked hash` -> `Copy cracked plaintext`